cmake/configs/modes.cmake

```diff
@@ -122,9 +122,9 @@ if(MSVC)
     set(asan_compiler_cxx_flag "/fsanitize=address")
     set(asan_lib_flags "")
 else()
-    set(asan_compiler_c_flag "-fsanitize=address -shared-libasan" )
-    set(asan_compiler_cxx_flag "-fsanitize=address -shared-libasan")
-    set(asan_link_flags "-fsanitize=address -shared-libasan")
+    set(asan_compiler_c_flag "-fsanitize=address" )
+    set(asan_compiler_cxx_flag "-fsanitize=address")
+    set(asan_link_flags "-fsanitize=address")
```

Comment:
I think this is fine for now, but we should think about how to address this when we think about merging into the develop branch.